apps/extension/src/core/domains/balances/handler.ts

```diff
@@ -89,107 +89,101 @@ const subscribeBalancesByParams = async (
   // create subscription callback
   const callback = createSubscription<"pri(balances.byparams.subscribe)">(id, port)
 
-  const obsSubscriptionParams = new BehaviorSubject<BalanceSubscriptionParams>({
+  // wait for chaindata to hydrate
+  await updateAndWaitForUpdatedChaindata()
+
+  // set up variables to track inner balances subscriptions
+  let subscriptionParams: BalanceSubscriptionParams = {
     addressesByTokenByModule: {},
     miniMetadataIds: [],
-  })
+  }
+  let balancesSubscriptionsGeneration = 0
+  let balancesUnsubCallbacks: Promise<UnsubscribeFn>[] = []
+
+  // create a subscription to the balances params
+  // allows us to restart the inner balances subscriptions whenever the data
+  // which they depend on changes
+  const byParamsSubscription = combineLatest([
+    // chains
+    chaindataProvider.chainsListObservable,
+    // evmNetworks
+    chaindataProvider.evmNetworksListObservable,
+    // tokens
+    chaindataProvider.tokensListObservable,
+    // miniMetadatas - not used here but we must retrigger the subscriptions when this changes
+    liveQuery(async () => await balancesDb.miniMetadatas.toArray()),
+
+    // active state of substrate chains
+    activeChainsStore.observable,
+    // active state of evm networks
+    activeEvmNetworksStore.observable,
```

Comment:
By changing the order of these, they can be passed into `getSubscriptionParams` just by doing `...args`